features/FEATURE_LWIP/TESTS/mbedmicro-net/host_tests/udp_shotgun.py

```diff
@@ -42,6 +43,9 @@ def handle(self):
             data = ''.join(map(chr, data))
             sock.sendto(data, self.client_address)
 
+            # Sleep a tiny bit to compensate for local network
+            time.sleep(0.001)
```

Comment:
Checked with @bridadan offline, looks like I had forgotten the MBED_STACK_STATS_ENABLED and MBED_HEAP_STATS_ENABLED, which cause increased delay on the device side.